modules/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/Init.java

```diff
@@ -102,6 +102,11 @@ public ConnectorFuture execute(Context context) {
         BStruct options = (BStruct) connector.getRefField(Constants.OPTIONS_STRUCT_INDEX);
         if (options != null) {
             populateSenderConfigurationOptions(senderConfiguration, options);
+            long maxActiveConnections = options.getIntField(Constants.MAX_ACTIVE_CONNECTIONS_INDEX);
+            if (!isInteger(maxActiveConnections)) {
```

Comment:
Since "maxActiveConnections" is a user input, Don't we need zero value and negative value checks?

Reply:
This is not an issue. Negative values are interpreted by the underlying library as  any-num-of-connections-allowed. And 0 is a valid value. If 0 is provided, no connections are created.